zipline/utils/sentinel.py

```diff
@@ -27,14 +41,14 @@ def __deepcopy__(self, _memo):
         def __copy__(self):
             return self
 
-    cls = type(result)
-    cls.__name__ = name
+    cls = type(Sentinel)
     try:
         # traverse up one frame to find the module where this is defined
-        cls.__module__ = sys._getframe(1).f_globals.get(
-            '__name__',
-            '__main__',
-        )
-    except (AttributeError, ValueError):
-        pass
-    return result
+        cls.__module__ = sys._getframe(1).f_globals['__name__']
+    except (AttributeError, ValueError, KeyError):
```

Comment:
`AttributeError` isn't needed so I will cut it. `ValueError` is when the stack depth is not deep enough to return the f_back (basically this would need to be the top frame). The keyerror is if `__name__` is not in the globals.